pyriemann/clustering.py

```diff
@@ -483,11 +489,231 @@ def _check_labels(self, X, y):
         return y
 
     def _get_z_score(self, d):
-        """get z-score from distance."""
+        """Get z-score from distance."""
         z = (d - self._mean) / self._std
         return z
 
     def _get_proba(self, z):
-        """get right-tailed proba from z-score."""
+        """Get right-tailed proba from z-score."""
         proba = 1 - norm.cdf(z)
         return proba
+
+
+class PotatoField(BaseEstimator, TransformerMixin, ClassifierMixin):
+
+    """Artefact detection with the Riemannian Potato Field.
+
+    The Riemannian Potato Field [1]_ is a clustering method used to detect
+    artifact in EEG signals. The algorithm combines several potatoes of low
+    dimension, each one being designed to capture specific artifact typically
+    affecting specific subsets of channels and/or specific frequency bands.
+
+    Parameters
+    ----------
+    n_potatoes : int (default 1)
+        Number of potatoes in the field.
+    p_threshold : float (default 0.01)
+        Threshold on probability to being clean, in (0, 1).
+    metric : string (default 'riemann')
+        The type of metric used for centroid and distance estimation.
+    z_threshold : float (default 3)
+        Threshold on z-score of distance to the centroid.
+    n_iter_max : int (default 10)
+        The maximum number of iteration to reach convergence.
+    pos_label: int (default 1)
+        The positive label corresponding to clean data.
+    neg_label: int (default 0)
+        The negative label corresponding to artifact data.
+
+    Notes
+    -----
+    .. versionadded:: 0.2.8
+
+    See Also
+    --------
+    Potato
+
+    References
+    ----------
+    .. [1] Q. Barthélemy, L. Mayaud, D. Ojeda, M. Congedo, "The Riemannian
+        potato field: a tool for online signal quality index of EEG", IEEE
+        TNSRE, 2019.
+    """
+
+    def __init__(self, n_potatoes=1, p_threshold=0.01, metric='riemann',
+                 z_threshold=3, n_iter_max=10, pos_label=1, neg_label=0):
+        """Init."""
+        self.n_potatoes = int(n_potatoes)
+        self.p_threshold = p_threshold
+        self.metric = metric
+        self.z_threshold = z_threshold
+        self.n_iter_max = n_iter_max
+        self.pos_label = pos_label
+        self.neg_label = neg_label
+
+    def fit(self, X, y=None):
+        """Fit the potato field from covariance matrices, with iterative
+        outlier removal to obtain reliable potatoes.
+
+        Parameters
+        ----------
+        X : list of n_potatoes ndarray of shape (n_trials, n_channels,
+                n_channels), with same n_trials but different n_channels
+            list of ndarray of SPD matrices.
+        y : ndarray, shape (n_trials,) | None (default None)
+            Labels corresponding to each trial. If None, all trials are
+            considered as clean.
```

Comment:
otherwise what is considered as clean?

Reply:
Trials having labels `y` equal to `pos_label` are considered as clean (explained in the doc of Potato and PotatoField parameters).

Reply:
can you clarify this docstring accordingly? 🙏